src/Microsoft.VisualStudio.Threading.Tests.Shared/JoinableTaskTests.cs

```diff
@@ -3249,6 +3249,34 @@ public void PostToUnderlyingSynchronizationContextShouldBeAfterSignalJoinableTas
             });
         }
 
+        [StaFact]
+        public void JoinAsyncShouldCompleteWithoutUIThreadAfterCancellation()
+        {
+            var jt = this.asyncPump.RunAsync(async delegate {
+                await Task.Yield();
+            });
+            var cts = new CancellationTokenSource();
+            var joinTask = jt.JoinAsync(cts.Token);
+            cts.Cancel();
```

Comment:
It can't have completed. We're on an STA thread in this test and the jt yielded. It can't complete unless the message pump finishes its execution, which the test never does. So we're guaranteed to simulate an incompleted delegate.